rest-client/src/test/java/com/fwmotion/threescale/cms/ThreescaleCmsClientImplUnitTest.java

```diff
@@ -796,10 +796,79 @@ void publish() {
     }
 
     @Test
-    void delete() {
+    void delete_File() throws Exception{
+        // Given a CmsFile object with an ID already
+        CmsFile newFile = new CmsFile();
+        newFile.setId(16);
+        newFile.setPath("/file.jpg");
+        newFile.setSectionId(30);
+        newFile.setTags(Set.of("a", "b", "c"));
+        newFile.setDownloadable(true);
+
+        // When the interface code is called
+        threescaleCmsClient.delete(newFile);
+
+        // Then only the file should have been deleted
+        then(filesApi).should(only()).deleteFile(
+            eq(newFile.getId()));
+        then(sectionsApi).shouldHaveNoInteractions();
+        then(templatesApi).shouldHaveNoInteractions();
     }
 
     @Test
-    void testDelete() {
+    void delete_FileById() throws Exception{
+        // Given a CmsFile object with an ID already
+        CmsFile newFile = new CmsFile();
+        newFile.setId(16);
+        newFile.setPath("/file.jpg");
+        newFile.setSectionId(30);
+        newFile.setTags(Set.of("a", "b", "c"));
+        newFile.setDownloadable(true);
+
+        // When the interface code is called
+        threescaleCmsClient.delete(newFile.getType(), newFile.getId());
+
+        // Then only the file should have been deleted
+        then(filesApi).should(only()).deleteFile(
+            eq(newFile.getId()));
+        then(sectionsApi).shouldHaveNoInteractions();
+        then(templatesApi).shouldHaveNoInteractions();
+    }
+
+    @Test
+    void delete_SectionById() throws Exception {
+        // Given a CmsSection object with an ID already
+        CmsSection cmsSection = new CmsSection();
+        cmsSection.setParentId(30);
+        cmsSection.setId(null);
+        cmsSection.setSystemName("new");
+        cmsSection.setPath("/new");
+        cmsSection.setTitle("new_section");
+        cmsSection.setPublic(true);
+
+        // When the interface code is called
+        threescaleCmsClient.delete(cmsSection.getType(), cmsSection.getId());
```

Comment:
Same deal as with `CmsFile`, no need to build a whole object just to pull 2 pieces out.